File: Learnify.Domain/Entities/Course.cs
namespace Learnify.Domain.Entities
{
    public class Course : BaseEntity
    {
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public Guid? ProfessorId { get; set; }
        public User? Professor { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
    }
}

File: Learnify.UI/src/models/entities/Course.js
export class Course {
    constructor(
        id,
        title,
        description,
        professorId,
        professor = null,
        createdAt = new Date(),
        updatedAt = new Date(),
        enrollments = [],
        modules = [],
        quizzes = [],
        reviews = []
    ) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.professorId = professorId;
        this.professor = professor;
        this.createdAt = createdAt;
        this.updatedAt = updatedAt;
        this.enrollments = enrollments;
        this.modules = modules;
        this.quizzes = quizzes;
        this.reviews = reviews;
    }
}
